uefi/section.go

```diff
@@ -210,10 +210,10 @@ func NewSection(buf []byte, fileOrder int) (*Section, error) {
 		if typeSpec.Attributes&uint16(GUIDEDSectionProcessingRequired) == 0 {
 			// No processing required
 			var err error
-			s.Encapsulated = &TypedFirmware{
+			s.Encapsulated = []*TypedFirmware{{
 				Type: reflect.TypeOf(&Section{}).String(),
-			}
-			s.Encapsulated.Value, err = NewSection(buf[typeSpec.DataOffset:], 0)
+			}}
+			s.Encapsulated[0].Value, err = NewSection(buf[typeSpec.DataOffset:], 0)
```

Comment:
You should loop and create new sections just like in NewFile, rather than just creating the first section. I also have no idea how alignment works within this encapsulated section, they're probably aligned wrt to dataoffset.

Reply:
Done!